fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java

```diff
@@ -154,11 +158,18 @@ private void pushFhirJsonsToCos(InputStream in, int dataLength) throws Exception
             if (cosBucketPathPrefix != null && cosBucketPathPrefix.trim().length() > 0) {
                 itemName = cosBucketPathPrefix + "/" + ResourceTypes.get(chunkData.getIndexOfCurrentResourceType())
                             + "_" + chunkData.getPartNum() + ".ndjson";
+                if (isExportPublic) {
+                    // Set expiration time to 2 hours(7200 seconds).
+                    metadata.setExpirationTime(Date.from(Instant.now().plusSeconds(7200)));
```

Comment:
Is this the part that you mentioned only works on Amazon S3 and not COS?  Or that's something else?

Reply:
yes, correct. IBM COS doesn't support object level expiration time, it only supports bucket level with policy setting which is well documented in IBM COS document, but IBM COS doesn't complaint if you set it even though it doesn't take effect at all. 

I have moved the codes into a separated if (isExportPublic) to avoid confusing - the old codes worked for S3 without any problem which I verified using my Amazon acount and also verified it didn't work in IBM COS;  I think this is because the meta object was really used only when cosClient.putObject is called later, I moved it just for the purpose of avoiding confusing of why do we change the meta after the request is created with the meta as a parameter.  and still, it works for S3 only.

Reply:
Thanks for the explanation...mind adding a quick comment in the code to indicate that IBM COS doesn't honor this (but also doesn't fail on it)?

Reply:
sure, good point!

Reply:
done